src/OrchardCore/OrchardCore.ResourceManagement/ResourceManager.cs

```diff
@@ -421,6 +414,21 @@ private ResourceRequiredContext[] DoGetRequiredResources(string resourceType)
                 return;
             }
 
+            // Use any additional dependencies from the settings without mutating the resource that is held in a singleton collection.
+            List<string> dependencies = null;
+            if (resource.Dependencies != null)
+            {
+                dependencies = new List<string>(resource.Dependencies);
+                if (settings.Dependencies != null)
+                {
+                    dependencies = dependencies.Concat(settings.Dependencies).ToList();
```

Comment:
`AddRange` might be better here, rather than allocating a new list with `Concat` ?

But I am just making guesses

Reply:
Yes better, did it too quickly ;)